wp-content/plugins/AB-Content-Alterer/ab-content-alterer.php

```diff
@@ -2,10 +2,142 @@
+      'Settings Admin', 
+      'AB Content Alterer Settings', 
+      'manage_options', 
+      'ab-settings-admin', 
+      array($this, 'create_admin_page')
+    );
+  }
+
+  /**
+   * Options page callback
+   */
+  public function create_admin_page() {
+    // Set class property
+    $this->options = get_option('ab_alterer_option_group');
+    ?>
+    <div class="wrap">
+      <h1>AB Content Alterer</h1>
+      <form method="post" action="options.php">
+      <?php
+        // This prints out all hidden setting fields
+        settings_fields( 'ab_alterer_options' );
+        do_settings_sections( 'ab-settings-admin' );
+        submit_button();
+      ?>
+      </form>
+    </div>
+    <?php
+  }
+
+  /**
+   * Register and add settings
+   */
+  public function page_init() {        
+    register_setting(
+      'ab_alterer_options', // Option group
+      'ab_alterer_option_group', // Option name
+      array( $this, 'sanitize' ) // Sanitize
+    );
+
+    add_settings_section(
+      'setting_section_id', // ID
+      'Content Altering settings', // Title
+      array( $this, 'print_section_info' ), // Callback
+      'ab-settings-admin' // Page
+    );  
+
+    add_settings_field(
+      'alterFrom', // ID
+      'Altered Text:', // Title 
+      array( $this, 'alterFromCallback' ), // Callback
+      'ab-settings-admin', // Page
+      'setting_section_id' // Section           
+    );      
+
+    add_settings_field(
+      'alterTo', 
+      'Text to alter:', 
+      array( $this, 'alterToCallback' ), 
+      'ab-settings-admin', 
+      'setting_section_id'
+    );      
+  }
+
+  /**
+   * Sanitize each setting field as needed
+   *
+   * @param array $input Contains all settings fields as array keys
+   */
+  public function sanitize( $input ) {
+    $new_input = [];
+    if( isset( $input['alterFrom'] ) )
+      $new_input['alterFrom'] = sanitize_text_field( $input['alterFrom'] );
+
+    if( isset( $input['alterTo'] ) )
+      $new_input['alterTo'] = sanitize_text_field( $input['alterTo'] );
+
+    return $new_input;
+  }
+
+  /** 
+   * Print the Section text
+   */
+  public function print_section_info() {
+    print 'Enter your Content Altering settings below:';
+  }
+
+  /** 
+   * Get the settings option array and print one of its values
+   */
+  public function alterFromCallback() {
+    printf(
+      '<input type="text" id="alterFrom" name="ab_alterer_option_group[alterFrom]" value="%s" />',
+      isset( $this->options['alterFrom'] ) ? esc_attr( $this->options['alterFrom']) : ''
+    );
+  }
+
+  /** 
+   * Get the settings option array and print one of its values
+   */
+  public function alterToCallback() {
+    printf(
+      '<input type="text" id="alterTo" name="ab_alterer_option_group[alterTo]" value="%s" />',
+      isset( $this->options['alterTo'] ) ? esc_attr( $this->options['alterTo']) : ''
+    );
+  }
+}
+
+if(is_admin()) {
+  $my_settings_page = new abContentAlererSettingsPage();
+}
+
 /**

```

Comment:
AbAlternateContent class should be defined in a separate file. 

Also, consider using namespaces to make your class names more intuitive and simple. http://php.net/manual/en/language.namespaces.php

Example: namespace 
namespace ABContentPlugin
class: ContentAlternation (class name without a prefix)